### PR TITLE
feat: port to typescript

### LINKED ISSUE #34: Add TypeScript types
I currently have a PR open, DefinitelyTyped/DefinitelyTyped#45824, to add types for `@snyk/ruby-semver`. However, it would be nice if these could be maintained and shipped with the actual package, instead of under `@types/snyk__ruby-semver`.

I'm still hoping to get some feedback from the DefinitelyTyped maintainers on my implementation, but are you open to the idea?

### PR DESCRIPTION
**Pull request by FauxFaux**
Fixes #34.

Nothing outrageous here. Compiler isn't apocalyptically angry about anything.. unless you turn `strict` on, then it is.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
+    "outDir": "./dist",
+    "target": "es2017",
+    "lib": [
+      "es2017"
```
Comment (gitphill) [inline tsconfig.json:6]: our lib template https://github.com/snyk/typescript-library uses es2018 - any reason not to do that here?

File: package.json [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
   "devDependencies": {
-    "ava": "^0.25.0"
+    "@ava/babel": "^1.0.1",
+    "@typescript-eslint/eslint-plugin": "^3.5.0",
+    "@typescript-eslint/parser": "^3.5.0",
+    "ava": "^3.9.0",
+    "eslint": "^6.8.0",
+    "eslint-config-prettier": "^6.11.0",
+    "prettier": "^2.0.5",
+    "typescript": "~3.8.3"
```
Comment (gitphill) [inline package.json:24]: can we not use latest? 3.9.5

File: tsconfig.json (hunk 1 above)
Comment (FauxFaux) [reply tsconfig.json:6]: This is CLI, so node 8. https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping#node-8 .  Your link is private, not CLI.

File: package.json (hunk 2 above)
Comment (FauxFaux) [reply package.json:24]: Typescript do not follow semver, and we have an issue with 3.9 in other libraries around https://github.com/microsoft/TypeScript/issues/38568 .  *Probably* doesn't matter here, but.

**Pull request merged**